src/darwin/Framework/CHIPTests/MTRSwiftPairingTests.swift

```diff
@@ -0,0 +1,88 @@
+import Matter
+import XCTest
+
+// This more or less parallels the "no delegate" case in MTRPairingTests
+
+struct Constants {
+    static let localPort = 5541
+    static let vendorID = 0xFFF1
+    static let onboardingPayload = "MT:Y.K90SO527JA0648G00"
+    static let deviceID = 0x12344321
+    static let timeoutInSeconds : UInt16 = 3
+}
+
+class MTRSwiftPairingTestControllerDelegate : NSObject, MTRDeviceControllerDelegate {
+    let expectation: XCTestExpectation
+
+    init(withExpectation providedExpectation: XCTestExpectation) {
+        expectation = providedExpectation
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, statusUpdate status: MTRCommissioningStatus) {
+        XCTAssertNotEqual(status, MTRCommissioningStatus.failed)
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, commissioningSessionEstablishmentDone error: Error?) {
+        XCTAssertNil(error)
+
+        do {
+            try controller.commissionNode(withID: Constants.deviceID as NSNumber, commissioningParams: MTRCommissioningParameters())
+        } catch {
+            XCTFail("Could not start commissioning of node")
+        }
+
+        // Keep waiting for commissioningComplete
```

Comment:
Since we might be using this code for illustration purposes as well maybe think about adding some more comments  about the flow between commissioning the node and the commissioningComplete callback.

Reply:
Oh, the illustration bit is going to be an edited version of this, with various text around it, etc.  It will have more better error handling than just "fail the test on anything unexpected".